src/kvilib/system/KviSignalHandler.cpp

```diff
@@ -0,0 +1,109 @@
+//=============================================================================
+//
+//   File : KviSignalHandler.cpp
+//   Creation date : Sat 2016-07-09 07:32:25 UTC by Vladimir Panteleev
+//
+//   This file is part of the KVIrc IRC client distribution
+//   Copyright (C) 1999-2010 Szymon Stefanek (pragma at kvirc dot net)
+//
+//   This program is FREE software. You can redistribute it and/or
+//   modify it under the terms of the GNU General Public License
+//   as published by the Free Software Foundation; either version 2
+//   of the License, or (at your option) any later version.
+//
+//   This program is distributed in the HOPE that it will be USEFUL,
+//   but WITHOUT ANY WARRANTY; without even the implied warranty of
+//   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+//   See the GNU General Public License for more details.
+//
+//   You should have received a copy of the GNU General Public License
+//   along with this program. If not, write to the Free Software Foundation,
+//   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+//
+//=============================================================================
+
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include "kvi_settings.h"
+
+#if !defined(COMPILE_ON_WINDOWS) && !defined(COMPILE_ON_MINGW)
+
+#include "KviSignalHandler.h"
+
+#include <signal.h>
+#include <sys/signal.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include <QApplication>
+
+// Adapted from http://doc.qt.io/qt-5/unix-signals.html
+
+// In the KviSignalHandler constructor, use the socketpair(2) function
+// to initialize the file descriptor pair, and then create the
+// QSocketNotifier to monitor the read end of the pair. The
+// activated() signal of each QSocketNotifier is connected to the
+// appropriate slot function, which effectively converts the Unix
+// signal to the QSocketNotifier::activated() signal.
+
+KviSignalHandler::KviSignalHandler(QObject *parent)
+        : QObject(parent)
+{
+    if (::socketpair(AF_UNIX, SOCK_STREAM, 0, fd))
+       qFatal("Couldn't create socketpair");
+
+    sn = new QSocketNotifier(fd[1], QSocketNotifier::Read, this);
+    connect(sn, SIGNAL(activated(int)), this, SLOT(handleSignal()));
+}
+
+// Somewhere else in your startup code, you install your Unix signal
+// handlers with sigaction(2).
+
+bool kvi_signalHandlerSetup()
+{
+	new KviSignalHandler();
+
+	struct sigaction sa;
+
+    sa.sa_handler = KviSignalHandler::unixSignalHandler;
```

Comment:
When you do catch up after holidays and come to this bit, a nice touch would be a rebase against master when youre at it. ;)

so then can finally merge this.